### PR TITLE
feat: Add sign in form, handle BE errors, display NavLinks based on isLoggedIn value

### PR DESCRIPTION
**Pull request by ElenaBezro**
- [x] add sign in form
- [x] display backend sign in and sign up errors
- [x]  display NavLinks based on isLoggedIn value

### CODE REVIEW

File: management-app/src/auth/SignUpForm/SignUpForm.tsx [hunk 1]
```diff
@@ -46,28 +47,38 @@ export function SignUpForm() {
...
-        // eslint-disable-next-line no-console
-        (e) => console.error(e)
-      );
+      dispatch(signUp({ name, login, password }))
+        .unwrap()
+        .then(() => {
+          setSignUpError('');
+          dispatch(signIn({ login, password }));
+        })
+        .catch((e) => setSignUpError(e.message));
```
Comment (ElenaBezro) [inline management-app/src/auth/SignUpForm/SignUpForm.tsx]:
```suggestion
      dispatch(signUp({ name, login, password }))
        .unwrap()
        .then(
          () => {
            setSignUpError('');
            return dispatch(signIn({ login, password }))
              .unwrap()
              .catch((e) => {
                // TODO: show user sign in error
              });
          },
          (e) => setSignUpError(e.message)
        );
```

File: management-app/src/components/Header/Header.tsx [hunk 2]
```diff
@@ -1,20 +1,42 @@
...
+          : ROUTES.filter(
+              (el) =>
+                !el.isShownWhenLoggedIn &&
+                el.id !== RouteID.NotFound &&
+                el.id !== RouteID.Welcome
+            ).map((el) => (
+              <NavLink key={el.id} to={el.routePath}>
+                {el.title}
+              </NavLink>
+            ))}
```
Comment (ElenaBezro) [inline management-app/src/components/Header/Header.tsx]:
```suggestion
        {ROUTES
          .filter((el) => el.isShownWhenLoggedIn === isLoggedIn
              && el.id !== RouteID.NotFound &&
              && el.id !== RouteID.Welcome
          ).map((el) => (
            <NavLink key={el.id} to={el.routePath}>
              {el.title}
            </NavLink>
          ))
        }
```

File: management-app/src/types.ts [hunk 3]
```diff
@@ -3,12 +3,15 @@ export type TPath = {
   routePath: string;
   title: string;
   element: JSX.Element;
+  isShownWhenLoggedIn: boolean;
 };
 
 export enum RouteID {
```
Comment (ElenaBezro) [inline management-app/src/types.ts:9]:
```suggestion
export enum RouteID {
```
```suggestion
export enum ERouteID {
```

File: management-app/src/components/Header/Header.tsx (hunk 2 above)
Comment (MarinaPresmytskaia) [reply management-app/src/components/Header/Header.tsx]: Great job!!!

**Review by MarinaPresmytskaia — APPROVED** [review]

**Review by treyer — APPROVED** [review]

**Pull request merged**